### PR TITLE
fix typo: 'git filter-repo' -> 'git-filter-repo'

### PR DESCRIPTION
**Pull request by KorayUlusan**
Presumably a typo. 

`git` doesn't have a `filter-repo` option. However `git-filter-repo` is a program (as linked in the docs).

### Why:

explained.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@KorayUlusan Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Comment by isaacmbrown** [comment]
Hi @KorayUlusan, thanks for taking the time to contribute!

While you're right that `git filter-repo` isn't a native Git command, it seems to be the correct syntax per the `git-filter-repo` README: https://github.com/newren/git-filter-repo#solving-this-with-filter-repo, and will presumably work once you have `git-filter-repo` installed 🙂

File: content/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ If you create a new clone of the repository, you won't lose any of your Git hist
    $ cd REPOSITORY-NAME
    ```
 
-5. To filter out the subfolder from the rest of the files in the repository, run [`git filter-repo`](https://github.com/newren/git-filter-repo), supplying this information:
+5. To filter out the subfolder from the rest of the files in the repository, run [`git-filter-repo`](https://github.com/newren/git-filter-repo), supplying this information:
```
Comment (isaacmbrown) [inline content/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository.md:33]:
```suggestion
5. To filter out the subfolder from the rest of the files in the repository, install [`git-filter-repo`](https://github.com/newren/git-filter-repo), then run `git filter-repo` with the following arguments.
```

File: content/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ If you create a new clone of the repository, you won't lose any of your Git hist
    {% endwindows %}
   
    ```shell
-   $ git filter-repo --path FOLDER-NAME1/ --path FOLDER-NAME2/
+   $ git-filter-repo --path FOLDER-NAME1/ --path FOLDER-NAME2/
```
Comment (isaacmbrown) [inline content/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository.md:47]:
```suggestion
   $ git filter-repo --path FOLDER-NAME1/ --path FOLDER-NAME2/
```

**Review by isaacmbrown — APPROVED** [review]
Applied a suggestion to clear up the confusion

**Pull request merged**